rfd/0109-opsgenie-plugin.md

```diff
@@ -0,0 +1,109 @@
+# OPSGenie Integration RFD
+## Required approvers
+
+Engineering:
+Product:
+Security:
+
+## What
+
+This RFD proposes a plugin that allows Teleport to integrate with OpsGenie. This plugin will differ from existing plugins by being part of the Teleport binary directly.
+
+## Success criteria
+Users are able to configure teleport to automatically create alerts in OpsGenie from access requests.
+Alerts were chosen over incidents in OpsGenie for high priority alerts that indicate a service interruption.
+Users are also able to configure auto approval flows to be met under certain conditions. E.g when a requester is on-call 
+
+## Configuration UX
+
+The plugin will be configured using a toml file containing the required to interact with both Teleport access and the OpsGenie API.
+
+```
+[teleport]
+addr = "example.com:3025" # Teleport Auth Server GRPC API address
+client_key = "/var/lib/teleport/plugins/opsgenie/auth.key" # Teleport GRPC client secret key
+client_crt = "/var/lib/teleport/plugins/opsgenie/auth.crt" # Teleport GRPC client certificate
+root_cas = "/var/lib/teleport/plugins/opsgenie/auth.cas" # Teleport cluster CA certs
+
+[opsgenie]
+api_key = "key" # Opsgenie API Key
+
+[log]
+output = "stderr" # Logger output. Could be "stdout", "stderr" or "/var/lib/teleport/opsgenie.log"
```

Comment:
Similar to above comments - if the plugin is running in-process, I would expect it to use the same logging config as the rest of the process.